### ISSUE TITLE
Unit test gradients

### ISSUE THREAD

**Issue by andrewdanks** [andrewdanks/JuliaNet#9]
Should unit test backprop to ensure we're calculating gradients properly and don't break the gradient descent in the future.
